cudamapper/src/main.cu

```diff
@@ -370,15 +370,28 @@ void postprocess_and_write_thread_function(const int32_t device_id,
                                                 0.5);
             }
 
-            // write to output
+            // write to output either SAM or PAF format
             {
-                GW_NVTX_RANGE(profiler, "main::postprocess_and_write_thread::print_paf");
-                print_paf(overlaps,
-                          cigars,
-                          *application_parameters.query_parser,
-                          *application_parameters.target_parser,
-                          application_parameters.kmer_size,
-                          output_mutex);
+                GW_NVTX_RANGE(profiler, "main::postprocess_and_write_thread::print_function");
+                if (!application_parameters.sam_format)
+                {
+                    print_paf(overlaps,
+                              cigars,
+                              *application_parameters.query_parser,
+                              *application_parameters.target_parser,
+                              application_parameters.kmer_size,
+                              output_mutex);
+                }
+#ifdef GW_BUILD_HTSLIB
```

Comment:
same here, we will need some error mechanism to report if sam is selected but HTSLIB is not available

Reply:
see above reply